vision/product_search/README.md

```diff
@@ -0,0 +1,37 @@
+# Google Cloud Vision API Product Search Go example
```

Comment:
We've been discouraged from writing READMEs and instead pointing people directly to the documentation (rather than duplicating it).

Reply:
done